### PR TITLE
refactor login to make request client-side and handle login errors

### PR DESCRIPTION
**Pull request by xtina-starr**
Uses client side fetch in order to be able to capture and bubble up error message per issue https://github.com/artsy/reaction-force/issues/123.

screen shot 2017-04-21 at 6 10 49 pm

### CODE REVIEW

File: src/apps/loyalty/containers/login/index.tsx [hunk 1]
```diff
@@ -1,4 +1,6 @@
+import *  as fetch from "isomorphic-fetch"
```
Comment (mzikherman) [inline src/apps/loyalty/containers/login/index.tsx]: extra space after the `*`

File: src/apps/loyalty/containers/login/index.tsx [hunk 2]
```diff
@@ -60,9 +66,69 @@ const StyledOrText = styled.div`
...
+    fetch(this.props.form.url, options).then(res => {
+      if (res.status === 200) {
+        // redirect to '/inquiries'
+        window.location.assign(`${sharify.data.BASE_URL}/inquiries`)
+      } else {
+        this.setState({
+          error: decodeURI(res.url.split("error=")[1]),
+        })
+      }
+    })
```
Comment (mzikherman) [inline src/apps/loyalty/containers/login/index.tsx]: this should probably `catch` also, maybe just log the error (or even maybe put it in the error bar too?)

**Comment by xtina-starr** [comment]
@alloy @l2succes @yuki @mzikherman do y'all have any additional feedback for this PR?

**Comment by mzikherman** [comment]
This looks awesome to me! I don't really have any further comments.

Maybe try writing a spec to assert that the error state is rendered? Something like https://github.com/artsy/reaction-force/blob/d4d13c16a6991b39115511237099e447ce964767/src/apps/loyalty/containers/inquiries/__tests__/index.tsx#L53, where you can pass/set error state.

Other than that it looks awesome!

**Comment by alloy** [comment]
I agree with @mzikherman, some test coverage would be nice. Otherwise this looks good to me 👍 

Don’t forget to assign someone for the merge.

**Comment by l2succes** [comment]
Looks good! I agree with the feedback @mzikherman left about displaying an error message in the red box in case of a network/server error from the `fetch` call as we discussed offline.

File: src/apps/loyalty/containers/login/__tests__/index.tsx [hunk 3]
```diff
@@ -1,33 +1,98 @@
...
+    })
+
+    const mockRedirectTo = Login.prototype.redirectTo = (url: string) => {
+      expect(url).toEqual("/loyalty/inquiries")
+    }
+
+    const login = <Login form={formConfig} />
+    const renderedComponent = TestUtils.renderIntoDocument(login) as Login
+
+    TestUtils.Simulate.submit(TestUtils.findRenderedDOMComponentWithTag(renderedComponent, "form"))
```
Comment (alloy) [inline src/apps/loyalty/containers/login/__tests__/index.tsx]: Are you sure this test works as expected? Seeing as this test does not work async like the others, it is possible that your expectation on line 69 simply never runs, thus is a false positive.

Comment (l2succes) [inline src/apps/loyalty/containers/login/__tests__/index.tsx]: Yes, we've run the test suite multiple times and the expectation in our stubbed `redirectTo` method always run as it's overridden before the invocation.

Comment (alloy) [inline src/apps/loyalty/containers/login/__tests__/index.tsx]: But did you test that it would fail if the implementation did not work correctly?

**Pull request merged**